test/test_common/utility.h

```diff
@@ -260,35 +260,76 @@ class TestUtility {
   static std::vector<std::string> split(const std::string& source, const std::string& split,
                                         bool keep_empty_string = false);
 
+  /**
```

Comment:
https://github.com/envoyproxy/envoy/blob/3e0ee3c1529a463ee7fa64a820e7224836ddde11/source/common/protobuf/utility.h#L144 was recently added, this should do the job I think.

Reply:
I can use that, but to avoid the removal getting really expensive I'd have to iterate from the back.  I think that makes the code harder enough to read I'd prefer to leave as-is unless you have strong feelings about it?

Reply:
The template could probably easily be extended to have other container types if you are concerned about the `list` vs. `vector` ^^.